### PR TITLE
[api] Return pre-signed S3 url for images.

### PR DESCRIPTION
**Pull request by charlesg3**
**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*
PCLUSTER-3679

*Description of changes:*
[api] Return pre-signed S3 url for images.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/models/s3_bucket.py [hunk 1]
```diff
@@ -386,3 +388,35 @@ def _configure_bucket(cls, bucket: S3Bucket, e: AWSClientError):
...
+
+def create_s3_presigned_url(s3_uri, expiration=3600):
+    """Generate a presigned URL to share an S3 object.
+
+    :param s3_uri: s3 uri, e.g. s3://my.bucket/my.object
+    :param expiration: Time in seconds for the presigned URL to remain valid
+    :return: Presigned URL as string
+    """
+    s3_uri_info = parse_bucket_url(s3_uri)
+    print(s3_uri_info)
```
Comment (demartinofra) [inline cli/src/pcluster/models/s3_bucket.py:419]: I believe you forgot to remove this debug print

File: cli/src/pcluster/models/imagebuilder.py [hunk 2]
```diff
@@ -237,6 +238,13 @@ def config_url(self):
                     raise ImageBuilderActionError(f"Unable to get image {self.image_id} config url.")
         return self.__config_url
 
+    @property
+    def presigned_config_url(self):
+        """Return configuration file as a presigned URL."""
+        if not self.__presigned_config_url:
+            self.__presigned_config_url = create_s3_presigned_url(self.config_url)
+        return self.__presigned_config_url
```
Comment (demartinofra) [inline cli/src/pcluster/models/imagebuilder.py:246]: have you considered implementing this similarly to cluster without parsing the s3 url?
```
    @property
    def config_presigned_url(self) -> str:
        """Return a pre-signed Url to download the config from the S3 bucket."""
        return self.bucket.get_config_presigned_url(
            config_name=PCLUSTER_S3_ARTIFACTS_DICT.get("source_config_name"),
            version_id=self.stack.original_config_version,
        )
```

File: cli/src/pcluster/api/controllers/image_operations_controller.py [hunk 3]
```diff
@@ -206,7 +206,7 @@ def describe_image(image_id, region=None):
 def _image_to_describe_image_response(imagebuilder):
     return DescribeImageResponseContent(
         creation_time=to_iso_time(imagebuilder.image.creation_date),
-        image_configuration=ImageConfigurationStructure(url=imagebuilder.config_url),
+        image_configuration=ImageConfigurationStructure(url=imagebuilder.presigned_config_url),
```
Comment (demartinofra) [inline cli/src/pcluster/api/controllers/image_operations_controller.py:209]: we should not fail if the s3 bucket or config file cannot be found. 

In the cluster controller we are returning a NOT_AVAILABLE string
```
    config_url = "NOT_AVAILABLE"
    try:
        config_url = cluster.config_presigned_url
    except ClusterActionError as e:
        # Do not fail request when S3 bucket is not available
        LOGGER.error(e)
```

File: cli/src/pcluster/models/imagebuilder.py (hunk 2 above)
Comment (charlesg3) [reply cli/src/pcluster/models/imagebuilder.py]: The stack might not exist if the image has been built and the stack is deleted.

Comment (charlesg3) [reply cli/src/pcluster/models/imagebuilder.py]: (or am I mis-understanding something about the imagebuilder flow)

File: cli/src/pcluster/models/imagebuilder.py [hunk 4]
```diff
@@ -224,6 +225,7 @@ def s3_artifact_dir(self):
     @property
     def config_url(self):
         """Return configuration file S3 bucket url."""
+        print("getting config url...")
```
Comment (demartinofra) [inline cli/src/pcluster/models/imagebuilder.py:228]: debug print?

**Review by demartinofra — COMMENTED** [review]
LGTM! only a print to be removed

**Review by demartinofra — APPROVED** [review]

**Pull request merged**